receiver/k8seventsreceiver/receiver_test.go

```diff
@@ -0,0 +1,171 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8seventsreceiver
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func TestNewReceiver(t *testing.T) {
+	rCfg := createDefaultConfig().(*Config)
+	client := fake.NewSimpleClientset()
+	r, err := newReceiver(
+		componenttest.NewNopReceiverCreateSettings(),
+		rCfg,
+		consumertest.NewNop(),
+		client,
+	)
+
+	require.NoError(t, err)
+	require.NotNil(t, r)
+	require.NoError(t, r.Start(context.Background(), componenttest.NewNopHost()))
+	assert.NoError(t, r.Shutdown(context.Background()))
+
+	rCfg.Namespaces = []string{"test", "another_test"}
+	r1, err := newReceiver(
+		componenttest.NewNopReceiverCreateSettings(),
+		rCfg,
+		consumertest.NewNop(),
+		client,
+	)
+
+	require.NoError(t, err)
+	require.NotNil(t, r1)
+	require.NoError(t, r1.Start(context.Background(), componenttest.NewNopHost()))
+	assert.NoError(t, r1.Shutdown(context.Background()))
+}
+
+func TestHandleEvent(t *testing.T) {
+	rCfg := createDefaultConfig().(*Config)
+	client := fake.NewSimpleClientset()
+	sink := new(consumertest.LogsSink)
+	r, err := newReceiver(
+		componenttest.NewNopReceiverCreateSettings(),
+		rCfg,
+		sink,
+		client,
+	)
+	require.NoError(t, err)
+	require.NotNil(t, r)
+	recv := r.(*k8seventsReceiver)
+	recv.ctx = context.Background()
+	k8sEvent := getEvent()
+	recv.handleEvent(k8sEvent)
+
+	assert.Equal(t, sink.LogRecordCount(), 1)
+}
+
+func TestEventDrop(t *testing.T) {
```

Comment:
If you are changing this PR for some reason, a better name would be `TestDropEventsOlderThanStartupTime` or something like this. This way, we know why it's being dropped.

Reply:
There are no further comments, so, I'll merge this as is, but feel free to send in a follow-up PR changing the test name if you see fit.

Reply:
Hello @jpkrohling, as per your suggestion, I have raised another PR(#7419).

Please review and merge the same if everything looks good.